src/ripple/app/misc/ValidatorList.h

```diff
@@ -505,6 +505,50 @@ class ValidatorList
         return {quorum_, trustedSigningKeys_};
     }
 
+    /**
+     * get the trusted master public keys
+     * @return the public keys
+     */
+    hash_set<PublicKey>
+    getTrustedMasterKeys()
+    {
+        std::unique_lock<std::shared_timed_mutex> lock{mutex_};
```

Comment:
Should these be shared locks? (here and below).

Reply:
Here: I am using an existing mutex because the same field is protected with this mutex in other functions. 
Below: I am currently using this mutex as well. But I can have a new mutex if you prefer.

Reply:
Sorry, I somehow missed this comment. It's not the mutex I'm flagging, it's the lock type. I think this could be a `shared_lock` rather than a `unique_lock`. I.e. 
```
    hash_set<PublicKey>
    getTrustedMasterKeys()
    {
        std::shared_lock lock{mutex_};
        return trustedMasterKeys_;
    }
```
Side note: since c++-17, constructors can infer their template parameters, so we don't need the `<std::shared_timed_mutex>` parameter anymore on these locks.

Reply:
I see, thanks. Will change.

Reply:
Done.